ygen/protoelements.go

```diff
@@ -41,6 +41,14 @@ func (*genState) yangTypeToProtoType(args resolveTypeArgs) (mappedType, error) {
 		return mappedType{nativeType: "ywrapper.StringValue"}, nil
 	case yang.Ydecimal64:
 		return mappedType{nativeType: "ywrapper.Decimal64Value"}, nil
+	case yang.Yleafref:
+		// We look up the leafref in the schema tree to be able to
```

Comment:
In the case that we do this above the switch, I presumed that we'd just change the args struct to a new one, but then figured that this actually means that we have to duplicate things since there is a case that we have a leafref to a leafref. This seems to handle that more cleanly?

Reply:
i suggest handling recursion inside the function that resolves the leaf-ref. i assume in principle you could have arbitrary nesting of leaf-refs? is so it seems simpler just to handle it inside a function that keeps recursing until it hits a non-ref type.